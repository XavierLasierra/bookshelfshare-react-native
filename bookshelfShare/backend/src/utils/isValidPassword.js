const bcrypt = require("bcrypt");

function isValidPassword(password) {
  return bcrypt.compareSync(password, this.password);
}

module.exports = { isValidPassword };
