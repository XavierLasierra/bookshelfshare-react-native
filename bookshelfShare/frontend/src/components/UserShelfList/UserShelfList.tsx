import React from 'react';
import {
  FlatList,
  SafeAreaView, TextInput, TouchableOpacity, View
} from 'react-native';
import { useSelector } from 'react-redux';

import Header from '../Header/Header';
import ShelfListElement from '../ShelfListElement/ShelfListElement';

import SearchIcon from '../../assets/searchIcon.svg';
import ShelfIcon from '../../assets/shelfIcon.svg';
import globalStyles from '../../styles/global.styles';
import styles from './userShelfList.styles';

export default function UserShelfList() {
  const userShelves = useSelector((store:any) => store.userShelves);
  const { userData: { _id: loggedUserId } } = useSelector((store: any) => store.loggedUser);

  function renderShelves({ item }: any) {
    return (
      <ShelfListElement shelf={item} loggedUserId={loggedUserId} />
    );
  }

  return (
    <SafeAreaView style={globalStyles.mainContainer}>
      <Header Logo={ShelfIcon} />
      <View style={styles.userShelfListContainer}>
        <View style={globalStyles.thinInputContainer}>
          <TextInput
            style={globalStyles.thinInput}
            placeholder="Search in all your shelves"
          />
          <TouchableOpacity>
            <SearchIcon width={35} height={35} />
          </TouchableOpacity>
        </View>
        <FlatList
          showsVerticalScrollIndicator={false}
          data={userShelves}
          renderItem={renderShelves}
          keyExtractor={(item, index) => `shelf-${index}`}
        />
        <View style={styles.marginBottom} />
      </View>
    </SafeAreaView>
  );
}
