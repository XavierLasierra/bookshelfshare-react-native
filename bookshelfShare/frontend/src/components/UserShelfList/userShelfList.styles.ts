import { StyleSheet } from 'react-native';
import stylesConstants from '../../styles/styles.constants';

export default StyleSheet.create({
  userShelfListContainer: {
    paddingHorizontal: stylesConstants.padding.normal
  },
  marginBottom: {
    width: '100%',
    height: 130
  }
});
