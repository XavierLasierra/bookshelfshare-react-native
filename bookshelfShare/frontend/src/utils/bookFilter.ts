export function bookFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfListFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.bookData.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfLocationFilter(filterLocation: number[], books: any) {
  return books.filter(({ customInformation: { location } }: any) => JSON.stringify(location)
  === JSON.stringify(filterLocation));
}
