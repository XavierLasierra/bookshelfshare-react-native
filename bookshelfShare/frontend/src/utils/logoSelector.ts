import BookIcon from '../assets/bookIcon.svg';
import SearchIcon from '../assets/searchIcon.svg';
import ProfileIcon from '../assets/profileIcon.svg';

export default function logoSelector(logoName: string) {
  switch (logoName) {
    case 'SearchIcon':
      return SearchIcon;
    case 'BookIcon':
      return BookIcon;
    case 'ProfileIcon':
      return ProfileIcon;
    default:
      return BookIcon;
  }
}
