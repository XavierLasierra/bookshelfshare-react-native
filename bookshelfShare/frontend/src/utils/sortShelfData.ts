export default function sortShelfData(shelf: any) {
  return shelf?.books.reduce((acc: any, book: any) => {
    const foundShelfBox = acc.find(({ location }: any) => JSON.stringify(location)
         === JSON.stringify(book?.customInformation?.location));
    if (foundShelfBox) {
      foundShelfBox.books.push(book.bookData);
    } else {
      acc.push({
        location: book?.customInformation?.location,
        books: [book.bookData]
      });
    }
    return acc;
  }, []);
}
