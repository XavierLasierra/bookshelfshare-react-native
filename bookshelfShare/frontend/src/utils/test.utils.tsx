import React from 'react';
import { render as rtlRender } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import configureStore from '../redux/store';

function render(
  component: any,
  initialState: any
) {
  const store = configureStore(initialState);

  function Wrapper({ children }: any) {
    return (
      <Provider store={store}>
        {children}
      </Provider>
    );
  }
  return rtlRender(component, { wrapper: Wrapper });
}

// re-export everything
export * from '@testing-library/react-native';
// override render method
export { render };
