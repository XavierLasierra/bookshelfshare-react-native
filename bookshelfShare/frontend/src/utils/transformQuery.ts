interface Query {
    isbn?: string,
    inauthor?: string,
    intitle?: string,
    inpublisher?: string
}

export default function transformQuery(query: Query):string {
  const queryEntries = Object.entries(query);
  const url = queryEntries
    .reduce((acc, queryElement, index) => `${acc}${queryElement[0]}=${queryElement[1]}${index < queryEntries.length - 1 ? '+' : ''}`, '');
  return `${url}&key=${process.env.GOOGLE_API_KEY}`;
}
