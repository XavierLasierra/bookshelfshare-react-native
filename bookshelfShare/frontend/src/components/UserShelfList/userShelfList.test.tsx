import React from 'react';
import UserShelfList from './UserShelfList';
import {render, fireEvent} from '../../utils/test.utils';
import loggedUserMock from '../../mocks/loggedUser.mock';
import shelfMock from '../../mocks/shelf.mock';

const navigation = {
  push: jest.fn(),
};

describe('Given a UserShelfList component', () => {
  describe('When it is rendered', () => {
    let screen: any;
    beforeEach(() => {
      const initialState = {
        loggedUser: {userData: loggedUserMock.user},
        userShelves: [shelfMock],
      };
      screen = render(<UserShelfList navigation={navigation} />, initialState);
    });
    test('Then should match the snapshot', () => {
      expect(screen).toMatchSnapshot();
    });

    describe('And newShelfButton is pressed', () => {
      test('Then navigation.push should have been called', () => {
        const newShelfButton = screen.getByTestId('newShelfButton');
        fireEvent.press(newShelfButton);

        expect(navigation.push).toHaveBeenCalled();
      });
    });
  });
});
