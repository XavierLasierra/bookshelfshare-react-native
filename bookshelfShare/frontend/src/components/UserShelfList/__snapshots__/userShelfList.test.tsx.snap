// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a UserShelfList component When it is rendered Then should match the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#F5F5F5",
      "minHeight": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "backgroundColor": "#F5F5F5",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#D9D8C4",
          "borderBottomLeftRadius": 25,
          "justifyContent": "center",
          "padding": 10,
          "width": "100%",
        }
      }
    >
      <IconMock
        height={45}
        width={45}
      />
    </View>
    <View
      style={
        Object {
          "height": 0,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#D9D8C4",
            "height": 20,
            "width": 20,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#F5F5F5",
              "borderTopRightRadius": 25,
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "paddingHorizontal": 30,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 10,
          "width": "100%",
        }
      }
    >
      <TextInput
        maxLength={25}
        placeholder="Search in all your shelves"
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderColor": "#ECECEC",
            "borderRadius": 10,
            "borderWidth": 1,
            "fontSize": 16,
            "height": "100%",
            "paddingHorizontal": 20,
            "width": "85%",
          }
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={false}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <IconMock
          height={35}
          width={35}
        />
      </View>
    </View>
    <RCTScrollView
      ListFooterComponent={
        <TouchableOpacity
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 450,
              "padding": 30,
            }
          }
          testID="newShelfButton"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 100,
                  "elevation": 5,
                  "height": 60,
                  "justifyContent": "center",
                  "width": 60,
                },
                Object {
                  "backgroundColor": "#ECECEC",
                  "height": 40,
                  "marginRight": 20,
                  "width": 40,
                },
              ]
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 20,
                }
              }
            >
              +
            </Text>
          </View>
          <Text
            style={
              Object {
                "color": "#3E3C37",
                "fontSize": 16,
              }
            }
          >
            Create new bookshelf
          </Text>
        </TouchableOpacity>
      }
      data={
        Array [
          Object {
            "_id": "1",
            "books": Array [
              Object {
                "bookData": Object {
                  "authors": Array [
                    "George Orwell",
                  ],
                  "categories": Array [
                    "London (England)",
                  ],
                  "description": "Winston Smith is a worker at the Ministry of Truth, where he falsifies records for the party. Secretly subversive, he and his colleague Julia try to free themselves from political slavery but the price of freedom is betrayal.",
                  "format": "BOOK",
                  "images": Object {
                    "smallThumbnail": "smallThumbnail1",
                    "thumbnail": "thumbnail1",
                  },
                  "isbn": Object {
                    "ISBN10": "0451524934",
                    "ISBN13": "9780451524935",
                  },
                  "language": "en",
                  "pageCount": 190,
                  "publishedDate": "2009",
                  "publisher": "No publisher found",
                  "subtitle": "A Novel",
                  "title": "1984",
                },
                "bookIsbn": "1",
                "customInformation": Object {
                  "location": Array [
                    0,
                    0,
                  ],
                },
              },
              Object {
                "bookData": Object {
                  "authors": Array [
                    "Orwell",
                    "Another author",
                    "Last author",
                  ],
                  "categories": Array [
                    "London (England)",
                    "Fiction",
                    "Another",
                  ],
                  "description": "Winston Smith is a worker at the Ministry of Truth, where he falsifies records for the party. Secretly subversive, he and his colleague Julia try to free themselves from political slavery but the price of freedom is betrayal.",
                  "format": "BOOK",
                  "images": Object {
                    "smallThumbnail": "smallThumbnail2",
                  },
                  "isbn": Object {
                    "ISBN10": "0123456789",
                    "ISBN13": "4567765432123",
                  },
                  "language": "en",
                  "pageCount": 190,
                  "publishedDate": "date",
                  "publisher": "Debolsillo",
                  "subtitle": "Subtitle",
                  "title": "Title",
                },
                "bookIsbn": "2",
                "customInformation": Object {
                  "location": Array [
                    0,
                    0,
                  ],
                },
              },
            ],
            "name": "Shelf",
            "shelf": Array [
              2,
              2,
            ],
            "users": Array [
              Object {
                "_id": "1",
                "photo": "photo",
              },
              Object {
                "_id": "2",
                "photo": "photo",
              },
              Object {
                "_id": "3",
                "photo": "photo",
              },
              Object {
                "_id": "4",
                "photo": "photo",
              },
            ],
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={Array []}
      viewabilityConfigCallbackPairs={Array []}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#F9CB83",
                "borderRadius": 10,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginVertical": 10,
                "opacity": 1,
                "paddingHorizontal": 20,
                "paddingVertical": 10,
              }
            }
            testID="shelfDetailPageButton"
          >
            <IconMock
              height={35}
              width={35}
            />
            <Text
              style={
                Object {
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "letterSpacing": 1,
                  "width": 150,
                }
              }
            >
              Shelf
            </Text>
            <View
              style={
                Object {
                  "height": 65,
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row-reverse",
                    "justifyContent": "flex-end",
                  }
                }
              >
                <Image
                  source={
                    Object {
                      "uri": "photo",
                    }
                  }
                  style={
                    Object {
                      "borderRadius": 100,
                      "height": 30,
                      "width": 30,
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "fontSize": 16,
                      "marginRight": 5,
                    }
                  }
                >
                   
                  +
                  2
                </Text>
              </View>
              <Text
                style={
                  Object {
                    "fontSize": 12,
                    "fontWeight": "bold",
                    "textAlign": "right",
                  }
                }
              >
                2
                 
                books
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": 450,
                "opacity": 1,
                "padding": 30,
              }
            }
            testID="newShelfButton"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#FFFFFF",
                    "borderRadius": 100,
                    "elevation": 5,
                    "height": 60,
                    "justifyContent": "center",
                    "width": 60,
                  },
                  Object {
                    "backgroundColor": "#ECECEC",
                    "height": 40,
                    "marginRight": 20,
                    "width": 40,
                  },
                ]
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                +
              </Text>
            </View>
            <Text
              style={
                Object {
                  "color": "#3E3C37",
                  "fontSize": 16,
                }
              }
            >
              Create new bookshelf
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "height": 130,
        "width": "100%",
      }
    }
  />
</RCTSafeAreaView>
`;
